znai-reactjs/src/doc-elements/code-snippets/SimpleCodeSnippet.js

```diff
@@ -34,8 +34,13 @@ class SimpleCodeSnippet extends Component {
         this.state = {displayFully: !this.limitLines}
     }
 
+    // handles changes during preview
     componentWillReceiveProps(nextProps) {
         this.processProps(nextProps)
+
+        if (this.props.readMore && this.props.readMoreVisibleLines !== nextProps.readMoreVisibleLines) {
+            this.setState({displayFully: false})
+        }
```

Comment:
If there are several expanded code snippets on the visible page, what will this code do?

Reply:
`this.props` and `displayFully` are instance bound. It will collapse only changed ones.